src/WebJobs.Script/Description/PowerShell/PowerShellFunctionInvoker.cs

```diff
@@ -261,18 +261,33 @@ internal static List<string> GetModuleFilePaths(string rootScriptPath, string fu
         {
             List<string> modulePaths = new List<string>();
             string functionFolder = Path.Combine(rootScriptPath, functionName);
+            string rootModuleDirectory = Path.Combine(rootScriptPath, PowerShellConstants.ModulesFolderName);
             string moduleDirectory = Path.Combine(functionFolder, PowerShellConstants.ModulesFolderName);
-            if (Directory.Exists(moduleDirectory))
+            string[] searchDirectories = new string[] { rootModuleDirectory, moduleDirectory };
+            modulePaths.AddRange(AddToModulePaths(searchDirectories));
+            return modulePaths;
+        }
+
+        internal static List<string> AddToModulePaths(string[] directories)
+        {
+            List<string> paths = new List<string>();
+            for (int i = 0; i < directories.Length; i++)
             {
-                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
-                    PowerShellConstants.ModulesManifestFileExtensionPattern));
-                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
-                    PowerShellConstants.ModulesBinaryFileExtensionPattern));
-                modulePaths.AddRange(Directory.GetFiles(moduleDirectory,
-                    PowerShellConstants.ModulesScriptFileExtensionPattern));
+                string currentDirectory = directories[i];
+                if (Directory.Exists(currentDirectory))
+                {
+                    paths.AddRange(Directory.GetFiles(currentDirectory,
+                        PowerShellConstants.ModulesManifestFileExtensionPattern,
+                        SearchOption.AllDirectories));
```

Comment:
Kindly also add tests for the following,

1. nested folders
2. ensure that resolution in case of duplicates is as expected.